src/eth.rs

```diff
@@ -857,6 +856,33 @@ impl<'d, T> RawHandle for EthDriver<'d, T> {
     }
 }
 
+pub struct EthFrame {
+    buf: *mut u8,
+    len: u32,
+}
+
+unsafe impl Send for EthFrame {}
+
+impl EthFrame {
+    const unsafe fn new(buf: *mut u8, len: u32) -> Self {
+        Self { buf, len }
+    }
+
+    pub const fn as_slice(&self) -> &[u8] {
+        unsafe { core::slice::from_raw_parts(self.buf, self.len as _) }
+    }
+
+    pub fn as_mut_slice(&mut self) -> &mut [u8] {
```

Comment:
I don't think this is a good idea. Why do you need a mutable access to the buffer?

Reply:
In my use case, per the sta2eth example in esp-idf I want to change the MAC address and some other data before sending it on its way. Otherwise I would have to do another allocation (e.g. into a Vec) beforehand.
Why don't you think it's a good idea? What are the downsides? It seems to be used mutably in esp-idf.

Reply:
Yeah. You can only change the data as long as the new data is not overrunning the frame size that is passed to you, while the original buffer might be larger, yet you don't know how large it actually is. Anyway. Leave it mutable but then if you implement `Deref` also implement `DerefMut`.

Reply:
That shouldn't be an issue since you can't change the length of a slice, right?

Reply:
What shouldn't be an issue?

Reply:
You don't know the original length of the whole buffer, just the length of the frame I side it. So if your transformation s result in a longer ethernet frame, you can't do these in-place, is what I'm saying.

Reply:
> if your transformations result in a longer ethernet frame
How would one do this? Isn't slice length immutable? I guess you could get a `*mut u8` from the `&mut [u8]` but that's `unsafe`.